compiler/rustc_ast_passes/src/feature_gate.rs

```diff
@@ -325,8 +325,12 @@ impl<'a> Visitor<'a> for PostExpansionVisitor<'a> {
                     cfg_hide => doc_cfg_hide
                     masked => doc_masked
                     notable_trait => doc_notable_trait
-                    keyword => doc_keyword
                 );
+
+                if nested_meta.has_name(sym::keyword) {
+                    let msg = concat!("`#[doc(keyword)]` is meant for internal use only");
```

Comment:
Hmm, maybe just allow a custom message to be chosen with the macro? I worry this will fall out of sync with the macro.